src/Controller.ts

```diff
@@ -7,6 +7,8 @@ import { BootApp, STATE_RUNNING, STATE_INACTIVE } from "./BootApp";
 import { findJvm } from "@pivotal-tools/jvm-launch-utils";
 import * as path from "path";
 import { readAll } from "./stream-util";
+import opn = require('opn')
```

Comment:
don't use require, use import instead.

Reply:
I tried something like this before: `import * as opn from "opn";` 

The typescript compiler didn't seem to like that one. When I looked inside the `.d.ts` file for 'opn' it has a comment like this:

```
/* Note that ES6 modules cannot directly export callable functions.
 * This file should be imported using the CommonJS-style:
 *   import x = require('opn');
 */
```

So I just followed that advice and typescript was happy with that, and it seems to work.

If you want me to change it to something else, can you please tell me exactly what you want? Typescript import mechanics aren't 100% understood by me and I'm afraid I'm just not sure what else would work.

Reply:
ES6 is part of the standard now while CommonJS is not, and there're also some slight differences in behavior as far as I know. So usually I use import whenever possible. But here for the package `opn`:

1. By i) uninstall `@types/opn` and ii) remove `"strict": true` from tsconfig.json, I managed to make `import * as opn from "opn"` compiled. 

2. I also tried `import opn = require("opn")`, it works well with `@types/opn`  installed. 

I compared the compiled `Controller.js` files in 1) and 2), and find the two .js file are exactly the same. 

@andxu Can you provide more details and suggestion?

Reply:
Missing semicolon in the end of this line, suggest to add it back anyway.